crates/oxc_linter/src/rules/oxc/bad_array_method_on_arguments.rs

```diff
@@ -60,57 +57,50 @@ impl Rule for BadArrayMethodOnArguments {
         if !node.kind().is_specific_id_reference("arguments") {
             return;
         }
-        let Some(parent_node_id) = ctx.nodes().parent_id(node.id()) else {
+        let Some(parent) = ctx.nodes().parent_node(node.id()) else {
             return;
         };
-        let AstKind::MemberExpression(member_expr) = ctx.nodes().kind(parent_node_id) else {
+        if !matches!(
+            parent.kind(),
+            AstKind::MemberExpression(_) | AstKind::ComputedMemberExpression(_)
+        ) {
             return;
-        };
-        let Some(parent_node_id) = ctx.nodes().parent_id(parent_node_id) else {
-            return;
-        };
-        let AstKind::CallExpression(_) = ctx.nodes().kind(parent_node_id) else {
+        }
+        let member_expr = parent.kind();
+        let Some(grandparent) = ctx.nodes().parent_node(parent.id()) else {
             return;
         };
+        if matches!(member_expr, AstKind::ComputedMemberExpression(_)) {
+            let great_grandparent = ctx.nodes().parent_kind(grandparent.id());
+            let Some(AstKind::CallExpression(_)) = great_grandparent else {
+                return;
+            };
+        } else if !matches!(member_expr, AstKind::ComputedMemberExpression(_)) {
```

Comment:
I think the 2nd check is redundant. Could just be `} else {`.

Reply:
This should be fixed in #11767, as we don't need this code anymore. I think it was some clippy lint rule that made me write it initially.